src/population.cpp

```diff
@@ -244,6 +305,10 @@ INSTANTIATE_TEMPLATE_METHODS(uint32_t)
 INSTANTIATE_TEMPLATE_METHODS(int64_t)
 INSTANTIATE_TEMPLATE_METHODS(uint64_t)
 
+#ifdef __APPLE__
```

Comment:
what happens w/o this?  Also, not knowing anything about macOS, is __APPLE__ the best way to catch this?

Reply:
Apparently, `__APPLE__` is a good way to detect Mac OS. It's mentioned [here](https://developer.apple.com/library/archive/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html) with some notes not to redefine functions.

Dropping it will lead to `size_t` duplicating `uint64_t` errors on other platforms. Do you have a better solution in mind?

Reply:
I'm fine w/ the `__APPLE__` ifdef, just wanted to make sure it's the 'right' way.

Reply:
I think we could get around using macros by defining a "hidden function(s)" doing the instantiation. Then we could template this `#ifdef` out, and have a generic type-test if `size_t` and `uint64_t` are really different.

I may also be in favor of reducing macro usage.

Reply:
No need to do it for this PR.  I'm ok w/ limited macro usage, and am comfortable w/ it in this case - they're clear, localized, and non-fancy.